File: DSA/ARARTIK_TYAGI_590011633/DAY_6_QUESTION_3.py
def longest_consecutive_increasing(nums):
    if not nums:
        return 0
    max_len = 1
    curr_len = 1
    for i in range(1, len(nums)):
        if nums[i] == nums[i-1] + 1:
            curr_len += 1
            max_len = max(max_len, curr_len)
        else:
            curr_len = 1
    return max_len

# User Input
nums = list(map(int, input("Enter array to find longest consecutive increasing subarray (space-separated): ").split()))
print("Length of longest consecutive increasing subarray:", longest_consecutive_increasing(nums))

File: DSA/Ayush_Mishra_590014844/Day_6_Question_3.c
#include <stdio.h>
int longestConsecutiveSubarray(int nums[], int size) {
    if (size == 0) return 0;

    int max_len = 1, curr_len = 1;
    for (int i = 1; i < size; i++) {
        if (nums[i] == nums[i - 1] + 1) {
            curr_len++;
            if (curr_len > max_len)
                max_len = curr_len;
        } else {
            curr_len = 1;
        }
    }
    return max_len;
}

int main() {
    int nums[] = {1, 2, 2, 3, 4, 1, 2, 3, 4, 5};
    int size = sizeof(nums) / sizeof(nums[0]);
    printf("Length of longest consecutive increasing subarray: %d\n", longestConsecutiveSubarray(nums, size));
    return 0;
}
